zendesk/poller.go

```diff
@@ -0,0 +1,68 @@
+// Package zendesk simply provides a method which regularly calls
+// the zendesk incremental ticket API
+package zendesk
+
+import (
+	"github.com/adamar/ZeGo/zego"
+	"strconv"
+	"time"
+
+	"github.com/hudl/zendesk-exporter/ticketwriter"
+)
+
+const (
+	// Zendesk allows this endpoint to be hit no more than 10 times/minute
+	// so our minimum wait time should be 6 seconds
+	minIntervalSec = 6
+
+	// This may have to be tweaked to find a sweet spot
+	maxIntervalSec = 60
+)
+
+// A Poller contains all the information necessary to regularly
+// poll the zendesk API such as credentials and startTime
+type Poller struct {
+	Auth      zego.Auth
+	StartTime string
+	TickWrt   ticketwriter.TicketWriter
+
+	// This is to prevent duplicates during low activity times
+	PrevTickId uint64
+}
+
+// Poll continuously hits the IncrementalTicket API starting from
+// StartTime and working up in batches of 1000 tickets following the
+// guidelines located in Zendesk's documentation:
+// https://developer.zendesk.com/rest_api/docs/core/incremental_export#tickets
+func (p *Poller) Poll() {
+	for {
+		results, err := p.Auth.IncrementalTicket(p.StartTime)
+		if err != nil {
+			log.Error("Error when polling for zendesk tickets: %+v", err)
```

Comment:
Maybe include some metadata here, such as the start time and anything about the poller that might be useful.